### PR TITLE
fix TestListenerTimeoutReadWrite

### PR DESCRIPTION
**Pull request by faddat**
This is a fix to a frequently seen test failure, TestListenerTimeoutReadWrite.

Please evaluate it carefully, as I do wish to preserve the rigor of the original test and in this case I have not gone deeply through the underlying code.

closes: #1915

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: privval/socket_listeners_test.go [hunk 1]
```diff
@@ -97,19 +97,18 @@ func TestListenerTimeoutAccept(t *testing.T) {
 
 func TestListenerTimeoutReadWrite(t *testing.T) {
 	const (
-		// This needs to be long enough s.t. the Accept will definitely succeed:
```
Comment (melekes) [inline privval/socket_listeners_test.go:100]: I find these comments helpful. Would you mind putting them back?

**Review by melekes — COMMENTED** [review]
Thanks @faddat ❤️ one minor comment

File: privval/socket_listeners_test.go (hunk 1 above)
Comment (faddat) [reply privval/socket_listeners_test.go:100]: I don't mind at all :)

**Comment by faddat** [comment]
its good to go

File: privval/socket_listeners_test.go (hunk 1 above)
Comment (melekes) [reply privval/socket_listeners_test.go:100]: Maybe you forgot to push the changes?

Comment (faddat) [reply privval/socket_listeners_test.go:100]: I was hesitant to touch the repo at all:

https://github.com/cometbft/cometbft/discussions/1424

File: privval/socket_listeners_test.go [hunk 2]
```diff
@@ -104,33 +104,39 @@ func TestListenerTimeoutReadWrite(t *testing.T) {
 		// Note: this controls how long this test actually runs.
 		timeoutReadWrite = 10 * time.Millisecond
 	)
-	for _, tc := range listenerTestCases(t, timeoutAccept, timeoutReadWrite) {
-		go func(dialer SocketDialer) {
-			_, err := dialer()
+
+	for i := 0; i < 100; i++ {
```
Comment (melekes) [inline privval/socket_listeners_test.go:108]: Do we need to repeat 100 times?

Comment (faddat) [reply privval/socket_listeners_test.go:108]: image


(and I checked and it costs a few seconds)

Comment (melekes) [reply privval/socket_listeners_test.go:108]: But the same could be said about all tests. Does that mean we have to repeat them 100 times too?

Comment (faddat) [reply privval/socket_listeners_test.go:108]: I suppose your POV is also accurate.

It passes 100 rounds well.

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**